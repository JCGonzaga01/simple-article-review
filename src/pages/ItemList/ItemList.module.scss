@import "helpers/stylings/mixins";
@import "helpers/stylings/global";

.wrapper {
  @include align-center;
  align-items: flex-start;
  flex-wrap: wrap;

  & > div:last-child {
    margin-right: 0;
  }

  .itemSummary {
    @include media-query(pc) {
      margin-right: 33px;
    }
    margin-bottom: 71px;
    width: 682px;

    .imageSrc {
      @include media-query(sp) {
        width: 100%;
      }

      margin-bottom: 28px;
      width: 682px;
      height: 384px;

      .tag {
        background: $tagBg;
        color: $tag;
        display: inline-table;
        font-family: $fontType1;
        font-size: $fontM;
        font-weight: 400;
        letter-spacing: 2px;
        line-height: $lineheightM;
        padding: 15px 0 0 15px;
        text-align: right;
        transform: rotate(180deg);
        writing-mode: vertical-lr;
      }
    }

    .titleContainer {
      display: flex;
      flex-direction: column;
      text-align: left;
      max-width: 466px;

      .title {
        color: $title;
        font-family: $fontType1;
        font-size: $fontL;
        font-weight: 700;
        letter-spacing: 0px;
        line-height: $lineheightL;
        margin-bottom: 37px;
      }

      .viewDetailsContainer {
        @include align-center;
        cursor: pointer;
        justify-content: flex-start;
        margin: auto auto auto 0;

        .line {
          background: $link;
          border: none;
          height: 2px;
          margin: 0 16px 0 0;
          width: 32px;
        }

        .link {
          color: $link;
          font-family: $fontType1;
          font-size: $fontXS2;
          font-weight: 700;
          letter-spacing: 0px;
          line-height: $lineheightXS2;
          text-transform: uppercase;
        }
      }
    }
  }
}
