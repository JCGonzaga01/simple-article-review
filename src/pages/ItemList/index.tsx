import React from "react";
import { useHistory } from "react-router-dom";
import data from "_root/data.json";
import { viewDetailsLinkText } from "constants/global";
import styles from "./ItemList.module.scss";

const ItemList: React.FC = () => {
  const history = useHistory();

  const handleOnClickDetails = (id: number) => () => {
    history.push({
      pathname: "/item",
      search: `?id=${id}`,
      state: { id },
    });
  };

  return (
    <div className={styles.wrapper}>
      {data.map((item) => {
        const url = process.env.PUBLIC_URL + `/assets/itemlist/${item.thumb}`;
        return (
          <div key={item.id} className={styles.itemSummary}>
            <div className={styles.imageSrc} style={{ backgroundImage: `url(${url})` }}>
              <span className={styles.tag}>{item.tag}</span>
            </div>
            <div className={styles.titleContainer}>
              <span className={styles.title}>{item.title_long}</span>
              <div className={styles.viewDetailsContainer} onClick={handleOnClickDetails(item.id)}>
                <hr className={styles.line} />
                <span className={styles.link}>{viewDetailsLinkText}</span>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default ItemList;
