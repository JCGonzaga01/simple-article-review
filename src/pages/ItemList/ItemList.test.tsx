import React from "react";
import renderer from "react-test-renderer";
import ItemList from "./index";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("<ItemList />", () => {
  let wrapper: any;
  let instance: any;

  beforeEach(() => {
    wrapper = renderer.create(<ItemList />);
    instance = wrapper.root;
  });

  test("should mount component", () => {
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render details", () => {
    expect(instance.findByProps({ className: "wrapper" }).children).toHaveLength(4);
  });

  test("should be able to call details page", () => {
    const viewDetailsButton = instance.findAll(
      (element: any) => element.props.className === "viewDetailsContainer"
    );
    viewDetailsButton[0].props.onClick();

    expect(mockHistoryPush).toHaveBeenCalledWith({
      pathname: "/item",
      search: `?id=1001`,
      state: { id: 1001 },
    });
  });
});
