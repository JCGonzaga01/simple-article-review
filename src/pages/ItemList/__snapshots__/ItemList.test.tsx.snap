// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemList /> should mount component 1`] = `
<div
  className="wrapper"
>
  <div
    className="itemSummary"
  >
    <div
      className="imageSrc"
      style={
        Object {
          "backgroundImage": "url(/assets/itemlist/1001_thumb.jpg)",
        }
      }
    >
      <span
        className="tag"
      >
        Toga Group
      </span>
    </div>
    <div
      className="titleContainer"
    >
      <span
        className="title"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ut libero id auctor.
      </span>
      <div
        className="viewDetailsContainer"
        onClick={[Function]}
      >
        <hr
          className="line"
        />
        <span
          className="link"
        >
          View Case Study
        </span>
      </div>
    </div>
  </div>
  <div
    className="itemSummary"
  >
    <div
      className="imageSrc"
      style={
        Object {
          "backgroundImage": "url(/assets/itemlist/1002_thumb.jpg)",
        }
      }
    >
      <span
        className="tag"
      >
        Hungry Jacks
      </span>
    </div>
    <div
      className="titleContainer"
    >
      <span
        className="title"
      >
        Aenean vulputate augue sed lorem ultricies, a placerat mi dictum
      </span>
      <div
        className="viewDetailsContainer"
        onClick={[Function]}
      >
        <hr
          className="line"
        />
        <span
          className="link"
        >
          View Case Study
        </span>
      </div>
    </div>
  </div>
  <div
    className="itemSummary"
  >
    <div
      className="imageSrc"
      style={
        Object {
          "backgroundImage": "url(/assets/itemlist/1003_thumb.jpg)",
        }
      }
    >
      <span
        className="tag"
      >
        UNSW
      </span>
    </div>
    <div
      className="titleContainer"
    >
      <span
        className="title"
      >
        Nulla non ante ultrices, blandit velit non
      </span>
      <div
        className="viewDetailsContainer"
        onClick={[Function]}
      >
        <hr
          className="line"
        />
        <span
          className="link"
        >
          View Case Study
        </span>
      </div>
    </div>
  </div>
  <div
    className="itemSummary"
  >
    <div
      className="imageSrc"
      style={
        Object {
          "backgroundImage": "url(/assets/itemlist/1004_thumb.jpg)",
        }
      }
    >
      <span
        className="tag"
      >
        Osko
      </span>
    </div>
    <div
      className="titleContainer"
    >
      <span
        className="title"
      >
        Vivamus vitae elit viverra, imperdiet quam quis, fermentum justo
      </span>
      <div
        className="viewDetailsContainer"
        onClick={[Function]}
      >
        <hr
          className="line"
        />
        <span
          className="link"
        >
          View Case Study
        </span>
      </div>
    </div>
  </div>
</div>
`;
